File: Ruby/PowerEfficientSpaceStation.rb
#encoding:utf-8

module Deepspace

    require_relative "PowerEfficientSpaceStationToUI.rb"

    class PowerEfficientSpaceStation < SpaceStation 

        @@EFFICIENCYFACTOR = 1.1

        def self.newCopy(station)
            super(station)
        end

        def getUIversion
            PowerEfficientSpaceStationToUI.new(self)
        end

        def fire
            super*@@EFFICIENCYFACTOR
        end

        def protection
            super*@@EFFICIENCYFACTOR
        end

        def setLoot(loot)
            super(loot)
            if loot.efficient
                Transformation::GETEFFICIENT
            else 
                Transformation::NOTRANSFORM
            end
        end

    end # PowerEfficientSpaceStation

end # Deepspace

File: Java/src/deepspace/PowerEfficientSpaceStation.java
package deepspace;

/**
 *
 * @author Shao Jie Hu Chen
 * @author Mario Megías Mateo
 */
public class PowerEfficientSpaceStation extends SpaceStation {
    private static final float EFFICIENCYFACTOR = 1.10f;
    
    public PowerEfficientSpaceStation(SpaceStation station) {
        super(station);
    }
    
    @Override
    public PowerEfficientSpaceStationToUI getUIversion() {
        return new PowerEfficientSpaceStationToUI(this);
    }
    
    @Override
    public float fire() {
        return super.fire()*EFFICIENCYFACTOR;
    }
    
    @Override
    public float protection() {
        return super.fire()*EFFICIENCYFACTOR;
    }
    
    @Override
    public Transformation setLoot(Loot loot) {
        super.setLoot(loot);
        return Transformation.NOTRANSFORM;
    }
}
